### DISCUSSION TITLE
Error while Cross compiling for IMX8MP (Armv8-A)

### DISCUSSION THREAD

**Discussion by goudam06** [openssl/openssl#22126] [Q&A] [answered]
Please do NOT use issues to ask questions about OpenSSL.

Instead, please use the [Q&A category in Discussions](<https://github.com/openssl/openssl/discussions/new?category=q-a>)
to ask your question.
1.set the environment by 
`source /opt/fslc-xwayland/4.0/environment-setup-armv8a-fslc-linux`
2. configure 
    `./Configure --openssldir=/home/mysticp0004/Desktop/Linux_Study/rootfs/usr/local/ssl --prefix=/home/mysticp0004/Desktop/Linux_Study/rootfs/usr/local/ --with-zlib-include=/home/mysticp0004/Desktop/Linux_Study/rootfs/usr/local/include/ --with-zlib-lib=/home/mysticp0004/Desktop/Linux_Study/rootfs/usr/local/lib/ zlib-dynamic linux-x86_64`
3. Build 
   `make`
4 Error 
//opt/fslc-xwayland/4.0/sysroots/x86_64-fslcsdk-linux/usr/bin/perl.real "-I." -Mconfigdata "util/dofile.pl" "-oMakefile" include/crypto/bn_conf.h.in > include/crypto/bn_conf.h
/opt/fslc-xwayland/4.0/sysroots/x86_64-fslcsdk-linux/usr/bin/perl.real "-I." -Mconfigdata "util/dofile.pl" "-oMakefile" include/crypto/dso_conf.h.in > include/crypto/dso_conf.h
/opt/fslc-xwayland/4.0/sysroots/x86_64-fslcsdk-linux/usr/bin/perl.real "-I." -Mconfigdata "util/dofile.pl" "-oMakefile" include/openssl/asn1.h.in > include/openssl/asn1.h
/opt/fslc-xwayland/4.0/sysroots/x86_64-fslcsdk-linux/usr/bin/perl.real "-I." -Mconfigdata "util/dofile.pl" "-oMakefile" include/openssl/asn1t.h.in > include/openssl/asn1t.h
/opt/fslc-xwayland/4.0/sysroots/x86_64-fslcsdk-linux/usr/bin/perl.real "-I." -Mconfigdata "util/dofile.pl" "-oMakefile" include/openssl/bio.h.in > include/openssl/bio.h
Can't locate strict.pm in @INC (you may need to install the strict module) (@INC contains: . /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/site_perl/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/site_perl/5.34.1 /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/vendor_perl/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/vendor_perl/5.34.1 /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/5.34.1) at configdata.pm line 6.
BEGIN failed--compilation aborted at configdata.pm line 6.
Compilation failed in require.
BEGIN failed--compilation aborted.
Can't locate strict.pm in @INC (you may need to install the strict module) (@INC contains: . /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/site_perl/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/site_perl/5.34.1 /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/vendor_perl/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/vendor_perl/5.34.1 /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/5.34.1) at configdata.pm line 6.
BEGIN failed--compilation aborted at configdata.pm line 6.
Compilation failed in require.
BEGIN failed--compilation aborted.
make: *** [Makefile:2492: include/crypto/bn_conf.h] Error 2
make: *** Waiting for unfinished jobs....
make: *** [Makefile:2494: include/crypto/dso_conf.h] Error 2
Can't locate strict.pm in @INC (you may need to install the strict module) (@INC contains: . /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/site_perl/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/site_perl/5.34.1 /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/vendor_perl/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/vendor_perl/5.34.1 /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/5.34.1) at configdata.pm line 6.
BEGIN failed--compilation aborted at configdata.pm line 6.
Compilation failed in require.
BEGIN failed--compilation aborted.
make: *** [Makefile:2496: include/openssl/asn1.h] Error 2
Can't locate strict.pm in @INC (you may need to install the strict module) (@INC contains: . /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/site_perl/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/site_perl/5.34.1 /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/vendor_perl/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/vendor_perl/5.34.1 /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/5.34.1) at configdata.pm line 6.
BEGIN failed--compilation aborted at configdata.pm line 6.
Compilation failed in require.
BEGIN failed--compilation aborted.
make: *** [Makefile:2498: include/openssl/asn1t.h] Error 2
Can't locate strict.pm in @INC (you may need to install the strict module) (@INC contains: . /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/site_perl/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/site_perl/5.34.1 /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/vendor_perl/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/vendor_perl/5.34.1 /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/5.34.1/x86_64-linux /usr/local/oe-sdk-hardcoded-buildpath/sysroots/x86_64-fslcsdk-linux/usr/lib/perl5/5.34.1) at configdata.pm line 6.
BEGIN failed--compilation aborted at configdata.pm line 6.
Compilation failed in require.
BEGIN failed--compilation aborted.
make: *** [Makefile:2500: include/openssl/bio.h] Error 2

**Comment by nhorman** [comment]
You're missing the strict.pm module, install it using your distribution package manager, and you should be fine

**Comment by tom-cosgrove-arm** [comment, marked as answer]
In addition to Neil's observation that your Perl environment seems broken, you are targeting an NXP iMX8M, which is Arm Cortex-A53, Armv8-A architecture (confirmed by the "set the environment" step), however you have configured for `linux-x86_64`.

Instead, you should configure for cross-compilation and `linux-aarch64` (assuming your target is running Linux, and that you are indeed cross-compiling on something like a Linux X86_64 system).

Assuming you have `CROSS_COMPILE` in your environment such as

```
export CROSS_COMPILE=aarch64-linux-gnu-
```

you will need

```
./Configure linux-aarch64 --cross-compile-prefix=$CROSS_COMPILE ... other options ...
```

**Reply by goudam06** [reply]
Thanks for the support now its work for me
